lua/lspconfig/util.lua

```diff
@@ -161,12 +161,12 @@ M.path = (function()
 
   -- Iterate the path until we find the rootdir.
   local function iterate_parents(path)
-    path = uv.fs_realpath(path)
+    path = uv.fs_realpath(path) or path
     local function it(s, v)
       if not v then
         return
       end
-      if is_fs_root(v) then
+      if is_fs_root(v) or not uv.fs_realpath(path) then
```

Comment:
I don't think this works, at least from the perspective of #1045 -- with [Lua generic for](https://www.lua.org/pil/7.2.html), the iterator is called to get the initial value, so in this case we'd be deferring the `nil` return to here and the loop below still wouldn't enter.

Reply:
```suggestion
      if is_fs_root(v) or not uv.fs_realpath(v) then
```